src/extension.ts

```diff
@@ -22,13 +22,18 @@ export function activate(context: vscode.ExtensionContext) {
 
         let dir = path.dirname(uri.fsPath);
 
-        if(isWslBash(vscode.workspace.getConfiguration('terminal'))) {
-            // c:\workspace\foo to /mnt/c/workspace/foo
-            dir = dir.replace(/(\w):/, '/mnt/$1').replace(/\\/g, '/')
-        }
-
         let terminal = vscode.window.createTerminal();
         terminal.show(false);
+
+        switch(kindOfShell(vscode.workspace.getConfiguration('terminal'))) {
+            case "wslbash":
+                // c:\workspace\foo to /mnt/c/workspace/foo
+                dir = dir.replace(/(\w):/, '/mnt/$1').replace(/\\/g, '/')
+                break;
+            case "cmd":
+                terminal.sendText(dir.slice(0,2));
```

Comment:
A comment here would be good, also do you know if we need to do this for powershell?

Reply:
I amended the commit to add comments before the line.

This was not required for powershell, it handles different drive letters on its own.